app/Models/ShopProduct.php

```diff
@@ -62,8 +62,8 @@ public static function boot()
     public function formatToCurrency($value, $locale = 'en_US')
     {
         $formatter = new NumberFormatter($locale, NumberFormatter::CURRENCY);
-
-        return $formatter->formatCurrency($value, $this->currency_code);
+    
+        return $formatter->formatCurrency((float) $value, $this->currency_code);
```

Comment:
We cannot use float for currency calculations

Reply:
Hm, ok. But why? Could you please tell me in more detail, since the logs say that the float value is expected

```
[previous exception] [object] (TypeError(code: 0): NumberFormatter::formatCurrency(): Argument #1 ($amount) must be of type float, string given at /var/www/controlpanel/app/Models/ShopProduct.php:59)
[stacktrace]```